packages/dds/merge-tree/src/mergeTree.ts

```diff
@@ -1442,15 +1442,9 @@ export class MergeTree {
             ref.callbacks?.beforeSlide?.();
             const removedRef = segment.localRefs.removeLocalRef(ref);
             assert(ref === removedRef, 0x2f3 /* Ref not in the segment localRefs */);
-            if (!newSegment) {
-                // No valid segments (all nodes removed or not yet created)
-                ref.segment = undefined;
-                ref.offset = 0;
-            } else {
-                ref.segment = newSegment;
-                ref.offset = newSegoff.offset ?? 0;
+            if (newSegment) {
```

Comment:
any reason to not preserve the old `!newSegment` case? Seems like it can be done with `ref.link(undefined, 0, undefined)`.

Reply:
we had chatted a bit about this, in another case, that references need to maintain their segment and offset. In general, i think it's a useful property to maintain that where we can

Reply:
there is also a technical issue here, in that link is only available to the LocalReferenceCollection, as the class is no longer exposed. The above remove internally removes via link, keeping segment an offset for the reasons above